src/group/managed_group/mod.rs

```diff
@@ -699,14 +756,14 @@ impl<'a> ManagedGroup<'a> {
         self.active
     }
 
-    /// Sets a different `CredentialBundle`
-    pub fn set_credential_bundle(&mut self, credential_bundle: &'a CredentialBundle) {
-        self.credential_bundle = credential_bundle;
-    }
-
-    /// Returns own credential
-    pub fn credential(&self) -> &Credential {
-        &self.credential_bundle.credential()
+    /// Returns own credential. If the group is inactive, it returns a
+    /// `UseAfterEviction` error.
+    pub fn credential(&self) -> Result<Credential, ManagedGroupError> {
```

Comment:
Why do you need this to return a copy?

Reply:
Ok, for the sake of getting this landed and as you say this should chnage in future, please add a ToDo and issue number.